app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java

```diff
@@ -140,53 +140,29 @@ public void rememberEulaAccepted(Context context){
     }
 
     /**
-     * User SDK function to login
-     * @param serverUrl
+     * @param server
      * @param username
      * @param password
      */
-    public void loginToDhis(String serverUrl, String username, String password){
-        //Delegate real login attempt to parent in sdk
-        super.login(serverUrl, username, password);
-    }
-
-    /**
-     * Ask for EULA acceptance if this is the first time user login to the server, otherwise login
-     * @param serverUrl
-     * @param username
-     * @param password
-     */
-    @Override
-    public void login(String serverUrl, String username, String password) {
-        //This method is overriden to capture credentials data
-        this.serverUrl=serverUrl;
-        this.username=username;
-        this.password=password;
-
-
-        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
-        if (!sharedPreferences.getBoolean(getString(R.string.eula_accepted), false)) {
-            askEula(R.string.settings_menu_eula, R.raw.eula, LoginActivity.this);
-        } else {
-            loginToDhis(serverUrl, username, password);
+    public void loginToDhis(EditText server, EditText username, EditText password){
+        try {
+            String authority = getString(org.hisp.dhis.client.sdk.ui.bindings.R.string.authority);
+            String accountType = getString(org.hisp.dhis.client.sdk.ui.bindings.R.string.account_type);
+            Inject.createUserComponent(server.toString(), authority, accountType).inject(this);
```

Comment:
@idelcano I think the use of dagger here was something you already fixed in future PRs so I'm accepting temporarily this. Please let me know otherwise